akka-remote/src/main/scala/akka/remote/artery/tcp/ArteryTcpTransport.scala

```diff
@@ -159,8 +180,19 @@ private[remote] class ArteryTcpTransport(
               if (controlIdleKillSwitch.isDefined)
                 outboundContext.asInstanceOf[Association].setControlIdleKillSwitch(controlIdleKillSwitch)
 
-              Flow[ByteString].prepend(Source.single(TcpFraming.encodeConnectionHeader(streamId))).via(connectionFlow)
+              Flow[ByteString]
+                .prepend(Source.single(TcpFraming.encodeConnectionHeader(streamId)).map { header =>
+                  logConnected()
+                  header
+                })
```

Comment:
`.wireTap`

Reply:
ok, can use that. maybe it would be enough to put the logging above instead, where we have the `afr.loFreq`. It's anyway a `() =>`. My thought was that in the prepend it was more sure that it was fully connected, but that might be overkill and we can see it as an connection attempt instead.

Reply:
No objection to doing it on the prepend, just thought of wiretap as then encoding of that `map => side effect; return element` pattern but actually maybe that was bad advice as wiretap could miss the element here since it just drops if there is no demand yet when the element comes.

Reply:
moved this up and renamed to connect instead of connected